packages/xo-web/src/common/form/toggle.js

```diff
@@ -28,7 +28,13 @@ export default class Toggle extends Component {
     const { props } = this
 
     if (!props.disabled) {
-      props.onChange(!props.value)
+      // turning props.onChange into a promise to cover both sync and async onChange functions
+      Promise.resolve(props.onChange(!props.value)).catch(error => {
+        // ignore when undefined because it usually means that the action has been canceled
+        if (error !== undefined) {
+          throw error
+        }
+      })
```

Comment:
You're actually not catching synchronous errors that might be thrown by `onChange` there. If you want to catch both sync and async errors, you can simply use an `await`:
 ```suggestion
      try {
        await props.onChange(!props.value)
      } catch (err) {
        if (err !== undefined) {
          throw err
        }
      }
```

Reply:
Doing so would change the way `Toggle` works, as it used to not await its `onChange`.

Do you think it's safe doing so?

Reply:
Before this PR, synchronous errors were not caught. As `confirm` is asynchronous, I don't think we need to catch them. (at least not for the fix I'm doing. Maybe it makes sense to catch them if we want to do something with that error, and we should do it in another commit)

If `props.onChange(!props.value)` throws synchronously, it will throw before being wrapped in the `Promise.resolve`